### PR TITLE
docs: fix broken links in undici webpage

### PR DESCRIPTION
**Pull request by dancastillo**
## This relates to...

https://github.com/nodejs/undici/issues/3322
https://github.com/nodejs/undici/issues/3748

## Changes

Using `renderer` for `links` can format them to route to appropriate page and not break readmes in github repo.

Absolute links will need to be used updated them for links in `docs/docs/api/` instead of relative links

https://github.com/user-attachments/assets/a519cb04-08a9-4073-9e5e-ac75e32dd6f5

### Bug Fixes

https://github.com/nodejs/undici/issues/3322
https://github.com/nodejs/undici/issues/3748

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [x] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -32,7 +32,37 @@
...
+            href = href.includes('/docs/docs/') ? href.replace('/docs/', '/') : href;
+
+            // ignore paths /docs/api/ and /docs/best-practices/ in /docs/docs directory
+            const ignorePaths = ['/docs/api', '/docs/best-practices'];
+
+            // Check for /docs/ in the href and remove it if present
+            // Want to ignore paths that are in the ignorePaths array
+            if (!ignorePaths.some(path => href.includes(path))) {
+              href = href.includes('/docs/') ? href.replace('/docs/', '/') : href;
+            }
```
Comment (Uzlopak) [inline docs/index.html:51]:
```suggestion
            // Check for /docs/ in the href and remove it if present
            if (href.startsWith('/docs/')) {
              // ignore paths /docs/api/ and /docs/best-practices/ in /docs/docs directory
              if (/\/docs\/(?:api|best-practices)(?:\/|$)/.test(href) === false) {
              href = href.slice(5);
              }
            }
```

**Review by Uzlopak — CHANGES_REQUESTED** [review]
imho needs to be stricter, see suggestion

File: docs/index.html (hunk 1 above)
Comment (dancastillo) [reply docs/index.html]: Good call out, updated the checks to reflect the suggestion

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**